File: level1/약수의 개수와 덧셈/Solution.java
class Solution {
    public int solution(int left, int right) {
        int sum = 0;
        for (int j = left; j <= right; j++) {
            int count = 0;
            for (int i = 1; i <= j; i++) { // 범위값 left~right 약수의 개수 구하기
                if (j % i == 0) {
                    count++; // 약수의 개수
                }
            }
            if (count % 2 == 0) { // 약수의 개수가 짝수
                sum += j;
            } else
                sum -= j;
        }
        return sum;
    }
}

File: level1/약수의 개수와 덧셈/Solution.py
def divisible(number):
    cnt = 0
    for i in range(1, number+1):
        if (number % i == 0) : cnt += 1
    return cnt

def solution(left, right):
    sum = 0
    for i in range(left, right+1):
        sum += +i if divisible(i) % 2 == 0 else -i
    return sum
  
  # 약수가 홀수개인 모든 수는 제곱 수
  # int(i**0.5) == i**0.5
